src/librustc_typeck/astconv.rs

```diff
@@ -1303,10 +1321,14 @@ fn split_auto_traits<'a, 'b, 'gcx, 'tcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>,
     (auto_traits, trait_bounds)
 }
 
-fn check_type_argument_count(tcx: TyCtxt, span: Span, supplied: usize,
-                             ty_param_defs: &[ty::TypeParameterDef]) {
-    let accepted = ty_param_defs.len();
-    let required = ty_param_defs.iter().take_while(|x| !x.has_default).count();
+fn check_type_argument_count(tcx: TyCtxt,
+                             span: Span,
+                             supplied: usize,
+                             ty_param_defs: usize,
+                             ty_param_defs_without_default: usize)
+{
+    let accepted = ty_param_defs;
```

Comment:
Oh, `accepted` is another good name (wherever I said `ty_max_len`). We should standardize on something.